File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(s) != len(t):
            return False

        map = {}
        for i in s:
            map[i] = map.get(i, 0) + 1

        for i in t:
            if i not in map.keys():
                return False
            elif map[i] == 0:
                return False
            else:
                map[i] = map.get(i) - 1

        return True

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())
        {
            return false;
        }

        int[] counts = new int[26];
        for(int i = 0; i < s.length(); i++)
        {
            counts[s.charAt(i) - 'a'] += 1;
            counts[t.charAt(i) - 'a'] -= 1;
        }

        for(int i: counts)
        {
            if(i != 0)
            {
                return false;
            }
        }

        return true;
    }
}